arelle/plugin/validate/ESEF/Util.py

```diff
@@ -6,44 +6,54 @@
 @author: Workiva
 (c) Copyright 2022 Workiva, All rights reserved.
 '''
+from __future__ import annotations
 import os, json
+
+from arelle.ModelObject import ModelObject
 from .Const import esefTaxonomyNamespaceURIs
 from lxml.etree import XML, XMLSyntaxError
 from arelle.FileSource import openFileStream
 from arelle.UrlUtil import scheme
+from arelle.ModelManager import ModelManager
+from arelle.ModelXbrl import ModelXbrl
+from arelle.ValidateXbrl import ValidateXbrl
+from typing import Any, Union, cast
+from collections.abc import Callable
+
+_: Callable  # Handle gettext
 
 # check if a modelDocument URI is an extension URI (document URI)
 # also works on a uri passed in as well as modelObject
-def isExtension(val, modelObject):
+def isExtension(val: ValidateXbrl, modelObject: ModelObject | None) -> bool:
     if modelObject is None:
         return False
     if isinstance(modelObject, str):
         uri = modelObject
     else:
         uri = modelObject.modelDocument.uri
     return (uri.startswith(val.modelXbrl.uriDir) or
-            not any(uri.startswith(standardTaxonomyURI) for standardTaxonomyURI in val.authParam["standardTaxonomyURIs"]))
+            not any(uri.startswith(standardTaxonomyURI) for standardTaxonomyURI in val.authParam["standardTaxonomyURIs"]))  # type: ignore[attr-defined]
```

Comment:
Running `print(type(val))` suggests val is of type `ValidateXbrl`. `ValidateXbrl` doesn't have any attribute `authParam` though so I choose to ignore the error for now.

Reply:
We could also create a `Protocol` for the `ValidateXbrl` class that implements `authParam`. This would also be needed in other files in the plugin folder.